system/lib/html5/emscripten_wget.cpp

```diff
@@ -0,0 +1,49 @@
+#include <emscripten.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+extern "C" {
+
+void emscripten_wget(const char* url, const char* file) {
+  // Create the ancestor directories.
+  char* copy = strdup(file);
+  auto* c = copy;
+  while (*c) {
+    // Create any non-trivial (not the root "/") directory.
+    if (*c == '/' && c != copy) {
+      *c = 0;
+      int result = mkdir(copy, 0777);
+      *c = '/';
+      // Continue while we succeed in creating directories or while we see that
+      // they already exist.
+      if (result < 0 && errno != EEXIST) {
+        free(copy);
+        return;
+      }
+    }
+    c++;
+  }
+  free(copy);
```

Comment:
Can you split this out into `static` local function call `mkdirs` or something like that?